test/fixtures/external_api_clients.yml

```diff
@@ -309,4 +309,50 @@ mailchimp_plugin:
     end
 
     # at the end of the file we have to implicitly return the class 
-    SyncToMailchimp
\ No newline at end of file
+    SyncToMailchimp
+
+sync_attribute_to_api_namespace_plugin:
+  api_namespace: one
+  slug: sync-attribute-to-api-namespace
+  label: SyncAttributeToApiNamespace
+  enabled: true
+  metadata: {
+            'attribute_name': 'test_attribute',
+            'default_value': 'test_value',
+            'placeholder_value': 'Enter you new_attribute value here.',
+          }
+  model_definition: |
+    class SyncAttributeToApiNamespace
+      def initialize(parameters)  
+        @external_api_client = parameters[:external_api_client]
+        @api_namespace = @external_api_client.api_namespace
+      end
+
+      def start
+        # Fetching provided inputs (attribute_name, default_value)
+        attribute_name = @external_api_client.metadata["attribute_name"]
+        placeholder_value = @external_api_client.metadata["placeholder_value"] || ''
+        default_value = @external_api_client.metadata["default_value"] || ''
+
+        raise 'The provided attribute is already defined in the ApiNamespace' if @api_namespace.properties.keys.include?(attribute_name)
+
+        ActiveRecord::Base.transaction do
```

Comment:
@alis-khadka lets not do this in a transaction. This will hold a lock on the API Resources table for the duration of the run. That isn't ideal in a live system.